src/scriptdev2/scripts/eastern_kingdoms/naxxramas/boss_anubrekhan.cpp

```diff
@@ -132,15 +128,24 @@ struct boss_anubrekhanAI : public ScriptedAI
         DoCastSpellIfCan(m_creature, SPELL_DOUBLE_ATTACK, CAST_TRIGGERED | CAST_AURA_NOT_PRESENT);
     }
 
-    void MoveInLineOfSight(Unit* pWho) override
+    void OnAnubDoorGo (Player* pPlayer, Creature* pCreature)
     {
-        if (!m_bHasTaunted && pWho->GetTypeId() == TYPEID_PLAYER && m_creature->IsWithinDistInMap(pWho, 110.0f) && m_creature->IsWithinLOSInMap(pWho))
+        if (GO_ARAC_ANUB_DOOR == GO_STATE_ACTIVE)
```

Comment:
This is very ugly... And also doesn't work.
If you need to make the creature to say the intro when the door changes the state, you will need to do this via the instance script.

When the door state is changed here: https://github.com/cmangos/mangos-classic/blob/master/src/scriptdev2/scripts/eastern_kingdoms/naxxramas/instance_naxxramas.cpp#L288
then you should make send the AI event to the boss, using itself as the sender and receiver.

Also, I'm not sure why you don't like the way I implemented this, with the IsWithinDistInMap(). Technically the boss should start yelling when the player enters the room. The door should be already open, so I don't understand what is your logic behind it.

PS. the line below ```if (boss_anubrekhanAI* pAnub = dynamic_cast<boss_anubrekhanAI*>(pCreature->AI()))``` should be replaced simply with ```SendAIEvent(id, invoker, m_creature)```